### PR TITLE
rustdesk: 1.2.3 -> 1.3.0

### PR DESCRIPTION
**Pull request by hatch01**
## Description of changes

rustdesk: 1.2.3 -> 1.3.0

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ru/rustdesk/package.nix [hunk 1]
```diff
@@ -33,50 +33,61 @@
...
-    substituteInPlace Cargo.toml --replace-fail "fufesou" "rustdesk-org";
-    substituteInPlace libs/enigo/Cargo.toml --replace-fail "fufesou" "rustdesk-org";
-  '';
+  patches = [
+    # Currently, cargo.lock is broken, so we need to update it using a patch
+    (fetchpatch {
+      url = "https://github.com/rustdesk/rustdesk/pull/9171/commits/6a2bc35e542cdde5944501e1ef88b4005e7b3991.patch";
+      hash = "sha256-/MGHuk7icB8frSbyKsZsjBYXp+IyngdHR5pZu3SGSO8=";
+    })
+  ];
```
Comment (getchoo) [inline pkgs/by-name/ru/rustdesk/package.nix:90]: Why is this needed if Cargo.lock is already vendored? Any patches required for it should be applied ahead of time. Per [this](https://github.com/NixOS/nixpkgs/blob/aabb78cb7ade02af5b8d4b07b534531972a91c0f/doc/languages-frameworks/rust.section.md#importing-a-cargolock-file-importing-a-cargolock-file) section in our Rust documentation, you should probably be doing this instead

```suggestion
  postPatch = ''
    ln -s ${./Cargo.lock} Cargo.lock
  '';
```

**Review by getchoo — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ru/rustdesk/package.nix (hunk 1 above)
Comment (hatch01) [reply pkgs/by-name/ru/rustdesk/package.nix]: Ok yes this is way better with a ln.
without it, I get this error : 
```bash
building '/nix/store/yh8s0widi99xihyf7igz27rwch2pq9vn-cargo-vendor-dir.drv'...
ln: failed to create symbolic link '/nix/store/8hapv8ax8i7631is0wfsrhq93vkcsxbj-cargo-vendor-dir/core-foundation-sys-0.8.6/8ryg75d8l1wwd7aw5s4zy4qywgvkw0l3-core-foundation-sys-0.8.6': Permission denied
error: builder for '/nix/store/yh8s0widi99xihyf7igz27rwch2pq9vn-cargo-vendor-dir.drv' failed with exit code 1;
       last 1 log lines:
       > ln: failed to create symbolic link '/nix/store/8hapv8ax8i7631is0wfsrhq93vkcsxbj-cargo-vendor-dir/core-foundation-sys-0.8.6/8ryg75d8l1wwd7aw5s4zy4qywgvkw0l3-core-foundation-sys-0.8.6': Permission denied
       For full logs, run 'nix log /nix/store/yh8s0widi99xihyf7igz27rwch2pq9vn-cargo-vendor-dir.drv'.
error: 1 dependencies of derivation '/nix/store/9fw71vs9gvdjwd771ir2nnly8ddybdj0-rustdesk-1.3.0.drv' failed to build
```

Comment (getchoo) [reply pkgs/by-name/ru/rustdesk/package.nix]: This is probably caused by the Cargo.lock having two entires for `core-foundation-sys` as shown [here](https://github.com/NixOS/nixpkgs/blob/271ffd08a7c5cf41e9d367bbbf9e549fd166868c/pkgs/by-name/ru/rustdesk/Cargo.lock#L1234-L1246). See https://github.com/NixOS/nixpkgs/issues/183344

**Comment by Leixb** [comment]
Result of `nixpkgs-review pr 337324` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rustdesk</li>
  </ul>
</details>

**Review by Leixb — APPROVED** [review]

File: pkgs/by-name/ru/rustdesk/package.nix [hunk 2]
```diff
@@ -33,49 +33,56 @@
   alsa-lib,
   makeDesktopItem,
   copyDesktopItems,
+  fetchpatch,
```
Comment (Leixb) [inline pkgs/by-name/ru/rustdesk/package.nix:36]: This is not needed, right?

File: pkgs/by-name/ru/rustdesk/package.nix [hunk 3]
```diff
@@ -37,45 +37,51 @@
...
     };
   };
 
-  prePatch = ''
-    # Patch 404 repos
-    substituteInPlace Cargo.lock --replace-fail "fufesou" "rustdesk-org";
-    substituteInPlace Cargo.toml --replace-fail "fufesou" "rustdesk-org";
-    substituteInPlace libs/enigo/Cargo.toml --replace-fail "fufesou" "rustdesk-org";
+  postPatch = ''
+    ln -fs ${./Cargo.lock} Cargo.lock
```
Comment (alyssais) [inline pkgs/by-name/ru/rustdesk/package.nix:84]: This should have a comment explaining when we can stop doing this.
